frontend/src/metabase/visualizations/lib/apply_tooltips.js

```diff
@@ -106,194 +110,185 @@ function applyChartTooltips(
-          if (Array.isArray(d.key)) {
-            // scatter
-            if (d.key._origin) {
-              data = d.key._origin.row.map((value, index) => {
-                const col = d.key._origin.cols[index];
-                return {
-                  key: getColumnDisplayName(col),
-                  value: value,
-                  col,
-                };
-              });
-            } else {
-              data = d.key.map((value, index) => ({
-                key: getColumnDisplayName(cols[index]),
+        let data = [];
+        if (Array.isArray(d.key)) {
+          // scatter
+          if (d.key._origin) {
+            data = d.key._origin.row.map((value, index) => {
+              const col = d.key._origin.cols[index];
+              return {
+                key: getColumnDisplayName(col),
                 value: value,
-                col: cols[index],
-              }));
-            }
-          } else if (d.data) {
-            // line, area, bar
-            if (!isSingleSeriesBar) {
-              cols = series[seriesIndex].data.cols;
-            }
+                col,
+              };
+            });
+          } else {
+            data = d.key.map((value, index) => ({
+              key: getColumnDisplayName(cols[index]),
+              value: value,
+              col: cols[index],
+            }));
+          }
+        } else if (d.data) {
+          // line, area, bar
+          if (!isSingleSeriesBar) {
+            cols = series[seriesIndex].data.cols;
+          }
 
-            data = [
-              {
-                key: getColumnDisplayName(cols[0]),
-                value: d.data.key,
-                col: cols[0],
-              },
-              {
-                key: getColumnDisplayName(cols[1]),
-                value: isNormalized
-                  ? formatValue(d.data.value, {
-                      number_style: "percent",
-                      column: cols[1],
-                    })
-                  : d.data.value,
-                col: { ...cols[1] },
-              },
-            ];
+          data = [
+            {
+              key: getColumnDisplayName(cols[0]),
+              value: d.data.key,
+              col: cols[0],
+            },
+            {
+              key: getColumnDisplayName(cols[1]),
+              value: isNormalized
+                ? formatValue(d.data.value, {
+                    number_style: "percent",
+                    column: cols[1],
+                  })
+                : d.data.value,
+              col: { ...cols[1] },
+            },
+          ];
 
-            // NOTE: The below overcomplicated code is due to using index (i) of
-            // the element in the DOM, as returned by d3
-            // It would be much preferable to somehow get the row more directly
+          const seriesData = series[seriesIndex].data || {};
+          const rawCols = seriesData._rawCols;
+          const rowIndex = d.data.index;
```

Comment:
It doesn't look like `data.index` is getting populated. Where should this get set?

Reply:
I think it's set by dc.js or d3. I thought I had this working, but it's entirely possible it's broken (there are test failures, after all)

Reply:
Interestingly it appears to work for non-timeseries questions.

Reply:
It looks like it was failing on quantitative scales too?

![image](https://user-images.githubusercontent.com/691495/59139251-eba70200-895f-11e9-8917-a7360ec6ed0b.png)

With that question, `rowIndex` is defined when the x scale is ordinal but undefined for any other scale.

Does that mean dc.js is binding different data depending on what scale is set? I didn't see anything about this is in the docs, but maybe it's the index within the ordinal scale rather than the index into the data. Those are the same here, but would be a reason why other scales don't set `index`.